test/Tests/ApiSpec.cs

```diff
@@ -94,44 +67,134 @@ public async Task Fail_to_create_auction(string sample, int index)
         var response = await application.PostAuction(FirstAuctionRequest, AuthToken.Seller1);
+        application.SetTime(StartsAt.AddHours(2));
         var bidResponse = await application.PostBidToAuction(1, @"{""amount"":""VAC11""}", AuthToken.Buyer1);
-        var auctionResponse = await application.GetAuction(1, AuthToken.Seller1);
         var bidResponseString = await bidResponse.Content.ReadAsStringAsync();
+        application.SetTime(EndsAt.AddHours(2));
+        var auctionResponse = await application.GetAuction(1, AuthToken.Seller1);
         var stringContent = await auctionResponse.Content.ReadAsStringAsync();
         Assert.Multiple(() =>
         {
             Assert.Equal(HttpStatusCode.Created,response.StatusCode);
             Assert.Equal(HttpStatusCode.OK,bidResponse.StatusCode);
             Assert.Empty(bidResponseString);
             Assert.Equal(HttpStatusCode.OK, auctionResponse.StatusCode);
-            Assert.Equal(JToken.Parse(@"{
-                ""id"": 1,
-                ""startsAt"": ""2022-07-01T10:00:00.000Z"",
-                ""title"": ""Some auction"",
-                ""expiry"": ""2022-09-18T10:00:00.000Z"",
-                ""seller"": ""seller1@hotmail.com"",
-                ""currency"": ""VAC"",
-                ""bids"": [{
-                    ""amount"": ""VAC11"",
-                    ""bidder"": ""buyer1@hotmail.com""
-                }]}").ToString(Formatting.Indented),
+            Assert.Equal(WithBid(FirstAuctionResponse,"VAC11","buyer1@hotmail.com").ToString(Formatting.Indented),
                 JToken.Parse(stringContent).ToString(Formatting.Indented));
         });
     }
     [Fact]
     public async Task Cannot_find_unknown_auction()
     {
-        using var application = new ApiFixture<TAuth>(typeof(TAuth).Name+"_"+nameof(Cannot_find_unknown_auction)+".db");
+        using var application = new ApiFixture<TAuth>(GetType().Name+"_"+nameof(Cannot_find_unknown_auction)+".db");
         var auctionResponse = await application.GetAuction(99, AuthToken.Seller1);
         Assert.Equal(HttpStatusCode.NotFound, auctionResponse.StatusCode);
     }
     [Fact]
     public async Task Cannot_place_bid_on_unknown_auction()
     {
-        using var application = new ApiFixture<TAuth>(typeof(TAuth).Name+"_"+nameof(Cannot_place_bid_on_unknown_auction)+".db");
+        using var application = new ApiFixture<TAuth>(GetType().Name+"_"+nameof(Cannot_place_bid_on_unknown_auction)+".db");
         var bidResponse = await application.PostBidToAuction(1, @"{""amount"":""VAC11""}", AuthToken.Buyer1);
         Assert.Equal(HttpStatusCode.NotFound, bidResponse.StatusCode);
     }
+}
+public class ApiAsyncSpecJwtToken:ApiAsyncSpec<JwtApiAuth>{}
+public class ApiAsyncSpecMsClientPrincipal:ApiAsyncSpec<MsClientPrincipalApiAuth>{}
+public abstract class ApiAsyncSpec<TAuth>
+    where TAuth:IApiAuth, new()
+{
+    [Fact]
+    public async Task Create_auction_1()
```

Comment:
Almost the same flow be used to test sync version as well, only difference is OK instead of Accepted